src/Ast.ml

```diff
@@ -1512,13 +1520,70 @@ end = struct
       mem_cls_cl cls cl
       || Hashtbl.Poly.find_or_add memo (cls, cl) ~default:exposed_
 
+  and mark_parenzed_inner_nested_match exp =
+    let exposed_ () =
+      let continue subexp =
+        if not (parenze_exp (sub_exp ~ctx:(Exp exp) subexp)) then
+          mark_parenzed_inner_nested_match subexp ;
+        false
+      in
+      match exp.pexp_desc with
+      | Pexp_assert e
+       |Pexp_construct
+          ({txt= Lident "::"}, Some {pexp_desc= Pexp_tuple [_; e]})
+       |Pexp_construct (_, Some e)
+       |Pexp_ifthenelse (_, e, None)
+       |Pexp_ifthenelse (_, _, Some e)
+       |Pexp_lazy e
+       |Pexp_newtype (_, e)
+       |Pexp_open (_, _, e)
+       |Pexp_fun (_, _, _, e)
+       |Pexp_sequence (_, e)
+       |Pexp_setfield (_, _, e)
+       |Pexp_setinstvar (_, e)
+       |Pexp_variant (_, Some e) ->
+          continue e
+      | Pexp_let (_, _, e)
+       |Pexp_letexception (_, e)
+       |Pexp_letmodule (_, _, e) ->
+          continue e
+      | Pexp_extension (_, PStr [{pstr_desc= Pstr_eval (e, _)}]) -> (
+        match e.pexp_desc with
+        | Pexp_function cases | Pexp_match (_, cases) | Pexp_try (_, cases) ->
+            List.iter cases ~f:(fun case ->
+                mark_parenzed_inner_nested_match case.pc_rhs ) ;
+            true
+        | _ -> continue e )
+      | Pexp_function cases | Pexp_match (_, cases) | Pexp_try (_, cases) ->
+          List.iter cases ~f:(fun case ->
+              mark_parenzed_inner_nested_match case.pc_rhs ) ;
+          true
+      | Pexp_apply (_, args) -> continue (snd (List.last_exn args))
+      | Pexp_tuple es -> continue (List.last_exn es)
+      | Pexp_array _ | Pexp_coerce _ | Pexp_constant _ | Pexp_constraint _
+       |Pexp_construct (_, None)
+       |Pexp_extension _ | Pexp_field _ | Pexp_for _ | Pexp_ident _
+       |Pexp_new _ | Pexp_object _ | Pexp_override _ | Pexp_pack _
+       |Pexp_poly _ | Pexp_record _ | Pexp_send _ | Pexp_unreachable
+       |Pexp_variant (_, None)
+       |Pexp_while _ ->
+          false
+    in
+    let _ : bool =
+      Hashtbl.Poly.find_or_add memo_parenzed_inner_nested_match exp
+        ~default:exposed_
+    in
+    ()
```

Comment:
Just FMI, do you prefer this style over `... |> (ignore : bool -> _)` for any particular reason?

Reply:
none. Changed.